### PR TITLE
Sockets - Grace

### PR DESCRIPTION
**Pull request by gracemshea**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | Yes, loops, conditionals and the like.
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | My searches produced a lot of old info on JS. Ruby's docs also more comprehensive, so will have to find best resource for latest JS.
What was a challenge you were able to overcome on this assignment? | Using console.log to figure out where my bugs were.
What has been interesting and positive about learning a new programming language? | Everything. I still feel like I need a lot of practice with JS and have a lot to learn, but I have enjoyed learning a new language. I feel like it also reinforces and teaches me things about Ruby by comparison. 
What is something to focus on your learnings in JavaScript in the next week? | Just writing it more. Syntax. Figuring out how to debug.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,115 @@
...
+    // While there remain elements to shuffle...
+    while (0 !== currentIndex) {
+    // Pick a remaining element...
+      randomIndex = Math.floor(Math.random() * currentIndex);
+      currentIndex -= 1;
+
+      // And swap it with the current element.
+      temporaryValue = array[currentIndex];
+      array[currentIndex] = array[randomIndex];
+      array[randomIndex] = temporaryValue;
```
Comment (dHelmgren) [inline src/adagrams.js:15]: You seem to have pulled this from Stack Overflow! When taking a large chunk of code, it's good to credit and link it!

File: src/adagrams.js [hunk 2]
```diff
@@ -1,7 +1,115 @@
...
+    }
+
+    return array;
+  },
+
   drawLetters() {
-    // Implement this method for wave 1
+    const array = 'AAAAAAAAABBCCDDDDEEEEEEEEEEEEFFGGGHHIIIIIIIIIJKLLLLMMNNNNNNOOOOOOOOPPQRRRRRRSSSSTTTTTTUUUUVVWWXYYZ'.split('');
+    this.shuffle(array);
+    return array.slice(0, 10);
```
Comment (dHelmgren) [inline src/adagrams.js:24]: nice!

File: src/adagrams.js [hunk 3]
```diff
@@ -1,7 +1,115 @@
...
+      W: 4,
+      X: 8,
+      Y: 4,
+      Z: 10
+    };
+
+    let sum = 0;
+
+    word.split('').forEach((letter) => {
+      sum += scores[letter.toUpperCase()];
```
Comment (dHelmgren) [inline src/adagrams.js:81]: nice!

**Comment by dHelmgren** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | yes
Small commits with meaningful commit messages | yes, but I'd be happy seeing EVEN MORE
**Code Requirements** |
`drawLetters` method | good
Uses appropriate data structure to store the letter distribution | good
All tests for `drawLetters` pass | yes
`usesAvailableLetters` method | yes
All tests for `usesAvailableLetters` pass | yes
`scoreWord` method | good
Uses appropriate data structure to store the letter scores | yes
All tests for `scoreWord` pass | yes
`highestScoreFrom` method | yes
Appropriately handles edge cases for tie-breaking logic | yes
All tests for `highestScoreFrom` pass | yes
**Overall** | Great work on this project! In this project you’ve taken some interesting logic and worked it into JavaScript syntax. I’m adding a few comments on places that this could be better and minor suggestions for improvement, but overall, you’ve done a great job of practicing syntax.

Dan's generic positive feedback
